packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,390 @@ const double _kHandleSize = 22.0;
+  bool _overflowOpen = false;
+
+  // Whether the overflow menu exists.
+  bool get _shouldShowMoreButton {
+    final int itemsInFirstMenu = _indexWhereOverflows == -1 ? _itemKeys.length : _indexWhereOverflows;
+    if (_itemKeys.isEmpty || itemsInFirstMenu == null) {
+      return false;
+    }
+    return itemsInFirstMenu < _itemKeys.length;
+  }
+
+  @override
+  void initState() {
+    _measureItemsNextFrame();
+    super.initState();
+  }
+
+  @override
+  void didUpdateWidget(_TextSelectionToolbar oldWidget) {
+    // If the widget has been updated, then the content in the menu could have
+    // changed, so it will be necessary to render another frame offscreen and
+    // re-measure.
+    if (widget.menuWidthChanged(oldWidget)) {
+      _menuContentWidth = null;
+      _measureItemsNextFrame();
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  // Measure the width of the container and how many items fit inside in order
+  // to decide which items to put in the overflow menu.
+  void _measureItemsNextFrame() {
+    SchedulerBinding.instance.addPostFrameCallback((Duration _) {
+      // If the menu is empty, no need to measure it.
+      if (_itemKeys.isEmpty) {
+        return;
+      }
+
+      assert(_containerKey.currentContext != null);
+      final RenderBox renderBoxContainer = _containerKey.currentContext.findRenderObject() as RenderBox;
+      double remainingContainerWidth = renderBoxContainer.size.width;
+
+      setState(() {
+        _indexWhereOverflows = _itemKeys.indexWhere((GlobalKey key) {
+          assert(key.currentContext != null);
+          final RenderBox renderBox = key.currentContext.findRenderObject() as RenderBox;
+
+          if (renderBox.size.width > remainingContainerWidth) {
+            return true;
+          }
+
+          remainingContainerWidth -= renderBox.size.width;
+          return false;
+        });
+
+        final RenderBox renderBoxMoreButton = _moreButtonKey.currentContext.findRenderObject() as RenderBox;
+        final double itemsWidth = renderBoxContainer.size.width - remainingContainerWidth;
+
+        _menuContentWidth = _shouldShowMoreButton
+          ? itemsWidth + renderBoxMoreButton.size.width
+          : itemsWidth;
+      });
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    final List<Widget> items = <Widget>[
-      if (handleCut != null) FlatButton(child: Text(localizations.cutButtonLabel), onPressed: handleCut),
-      if (handleCopy != null) FlatButton(child: Text(localizations.copyButtonLabel), onPressed: handleCopy),
-      if (handlePaste != null) FlatButton(child: Text(localizations.pasteButtonLabel), onPressed: handlePaste),
-      if (handleSelectAll != null) FlatButton(child: Text(localizations.selectAllButtonLabel), onPressed: handleSelectAll),
-    ];
+
+    final List<Widget> items = <Widget>[];
+    _itemKeys.removeRange(0, _itemKeys.length);
```

Comment:
Same as _itemKeys.clear()

Reply:
The code was removed with the RenderBox approach.